include/openthread/commissioner.h

```diff
@@ -119,6 +119,19 @@ OTAPI otError OTCALL otCommissionerAddJoiner(otInstance *        aInstance,
  */
 OTAPI otError OTCALL otCommissionerRemoveJoiner(otInstance *aInstance, const otExtAddress *aEui64);
 
+/**
+ * This function gets the Provisioning URL.
+ *
+ * @param[in]    aInstance             A pointer to an OpenThread instance.
+ * @param[out]   aProvisioningUrl      A pointer to return the Provisioning URL. Note that the URL string is not
+ *                                     necessarily null-terminated. The return value specifies the length of the URL
+ *                                     string.
+ *
+ * @returns The length of URL string (number of chars).
+ *
+ */
+uint16_t otCommissionerGetProvisioningUrl(otInstance *aInstance, const char **aProvisioningUrl);
```

Comment:
Is there a reason for the return value being the size rather than the buffer pointer?  Said differently, why obtain the buffer pointer via the argument?

Reply:
Not really. I agree reversing them (getting buffer pointer as return value and size as variable pointer) makes more sense. Will change it.